src/parquet/column/column-writer-test.cc

```diff
@@ -358,13 +364,16 @@ TYPED_TEST(TestPrimitiveWriter, RequiredVeryLargeChunk) {
   ASSERT_EQ(SMALL_SIZE, this->values_read_);
   this->values_.resize(SMALL_SIZE);
   ASSERT_EQ(this->values_, this->values_out_);
+  std::vector<Encoding::type> encodings = this->metadata_encodings();
+  // There are 3 encodings (RLE, PLAIN_DICTIONARY, PLAIN) in a fallback case
+  // Dictionary encoding is not allowed for boolean type
+  // There are 2 encodings (RLE, PLAIN) in a non dictionary encoding case
+  ASSERT_EQ(Encoding::RLE, encodings[0]);
   if (this->type_num() != Type::BOOLEAN) {
-    // There are 3 encodings (RLE, PLAIN_DICTIONARY, PLAIN) in a fallback case
-    ASSERT_EQ(3, this->metadata_num_encodings());
+    ASSERT_EQ(Encoding::PLAIN_DICTIONARY, encodings[1]);
```

Comment:
This comment to explicitly verify the encodings did uncover a bug in the metadata writer